src/targets/gpu/mlir.cpp

```diff
@@ -988,6 +1044,7 @@ std::string dump_mlir(const module& m, const std::vector<shape>& inputs)
         mr = &mm;
         adjust_param_shapes(mm, inputs);
     }
+    rewrite_reduce(mm);
```

Comment:
This wont dump correctly if the inputs are empty. Probably should take the module by value in the function and remove the `mm` and `mr` variables.

Reply:
> This wont dump correctly if the inputs are empty. 

Why is that ? 
`rewrite_reduce` is just reshaping reduction ops. It doesn't touch inputs.

Reply:
> > This wont dump correctly if the inputs are empty.
> 
> Why is that ? `rewrite_reduce` is just reshaping reduction ops. It doesn't touch inputs.

It rewrites the `mm` variable which is only set to the `m` variable(ie the module passed to the function) when the inputs are not empty.

Reply:
Done.